packages/collector/test/tracing/cloud/aws/s3/test.js

```diff
@@ -0,0 +1,200 @@
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.n).to.equal('node.http.server'),
+        span => expect(span.data.http.url).to.equal(apiPath)
+      ]);
+    }
+
+    function verifyS3Exit(spans, parent, withError) {
+      return expectExactlyOneMatching(spans, [
+        span => expect(span.n).to.equal('s3'),
+        span => expect(span.k).to.equal(constants.EXIT),
+        span => expect(span.t).to.equal(parent.t),
+        span => expect(span.p).to.equal(parent.s),
+        span => expect(span.f.e).to.equal(String(appControls.getPid())),
+        span => expect(span.f.h).to.equal('agent-stub-uuid'),
+        span => expect(span.error).to.not.exist,
+        span => expect(span.ec).to.equal(withError ? 1 : 0),
+        span => expect(span.async).to.not.exist,
+        span => expect(span.data).to.exist,
+        span => expect(span.data.s3).to.be.an('object'),
+        span => expect(span.data.s3.op).to.exist
+      ]);
+    }
+  });
+
+  describe('tracing disabled', () => {
+    this.timeout(config.getTestTimeout() * 2);
+
+    const appControls = new ProcessControls({
+      appPath: path.join(__dirname, 'app'),
+      port: 3215,
+      useGlobalAgent: true,
+      tracingEnabled: false,
+      env: {
+        AWS_S3_BUCKET_NAME: bucketName
+      }
+    });
+
+    ProcessControls.setUpHooksWithRetryTime(retryTime, appControls);
+
+    requestMethods.forEach(requestMethod => {
+      describe(`attempt to get result with: ${requestMethod}`, () => {
+        availableOperations.forEach(operation => {
+          it(`should not trace (${operation})`, async () => {
+            await appControls.sendRequest({
+              method: 'GET',
+              path: `/${operation}/${requestMethod}`
+            });
+            return retry(() => agentControls.getSpans(), retryTime)
+              .then(() => agentControls.getSpans())
+              .then(spans => {
+                if (spans.length > 0) {
+                  fail(`Unexpected spans (AWS S3 suppressed: ${stringifyItems(spans)}`);
+                }
+              });
+          });
+        });
+      });
+    });
+  });
+
+  describe('tracing enabled but suppressed', () => {
+    const appControls = new ProcessControls({
+      appPath: path.join(__dirname, 'app'),
+      port: 3215,
+      useGlobalAgent: true,
+      env: {
+        AWS_S3_BUCKET_NAME: bucketName
+      }
+    });
+
+    ProcessControls.setUpHooksWithRetryTime(retryTime, appControls);
+
+    requestMethods.forEach(requestMethod => {
+      describe(`attempt to get result with: ${requestMethod}`, () => {
+        availableOperations.forEach(operation => {
+          it(`should not trace (${operation})`, async () => {
+            await appControls.sendRequest({
+              method: 'GET',
+              path: `/${operation}/${requestMethod}`,
+              headers: {
+                'X-INSTANA-L': '0'
```

Comment:
Minor nitpick: The `sendRequest` method actually supports an attribute `suppressTracing` which should be used instead of "manually" adding `X-INSTANA-L: 0` as a header. I have probably not been consistent with that at all, so I can see where this is coming from.